server/handlers/genomic_annotations.go

```diff
@@ -1 +1,155 @@
+		return genomic_annotations.NewGetValuesDefault(500).WithPayload(&genomic_annotations.GetValuesDefaultBody{
+			Message: "Query execution error: " + err.Error(),
+		})
+	}
+	defer rows.Close()
+
+	var annotations []string
+	var annotation string
+
+	for rows.Next() {
+		err := rows.Scan(&annotation)
+		if err != nil {
+			logrus.Error("Query result reading error: " + err.Error())
+			return genomic_annotations.NewGetValuesDefault(500).WithPayload(&genomic_annotations.GetValuesDefaultBody{
+				Message: "Query result reading error: " + err.Error(),
+			})
+		}
+		annotations = append(annotations, annotation)
+	}
+
+	if len(annotations) > 0 {
+		return genomic_annotations.NewGetValuesOK().WithPayload(annotations)
+	} else {
+		return genomic_annotations.NewGetValuesNotFound()
+	}
+}
+
+func MedCoGenomicAnnotationsGetVariantsHandler(params genomic_annotations.GetVariantsParams, principal *models.User) middleware.Responder {
+
+	err := utilserver.DBConnection.Ping()
+	if err != nil {
+		logrus.Error("Impossible to connect to DB: " + err.Error())
+		return genomic_annotations.NewGetVariantsDefault(500).WithPayload(&genomic_annotations.GetVariantsDefaultBody{
+			Message: "Impossible to connect to DB: " + err.Error(),
+		})
+	}
+	zygosity := ""
+	if len(params.Zygosity) > 0 {
+		zygosity = params.Zygosity[0]
+
+		for i := 1; i < len(params.Zygosity); i++ {
+			zygosity += "|" + params.Zygosity[i]
+		}
+	}
+
+	var query string
+
+	query, err = buildGetVariantsQuery(params)
+	if err != nil {
+		logrus.Error("Query execution error " + err.Error())
+		return genomic_annotations.NewGetValuesDefault(500).WithPayload(&genomic_annotations.GetValuesDefaultBody{
+			Message: "Query execution error: " + err.Error(),
+		})
+	}
+
+	rows, err := utilserver.DBConnection.Query(query, params.Value, zygosity)
+	if err != nil {
+		logrus.Error("Query execution error" + err.Error())
+		return genomic_annotations.NewGetVariantsDefault(500).WithPayload(&genomic_annotations.GetVariantsDefaultBody{
+			Message: "Query execution error " + err.Error(),
+		})
+	}
+	defer rows.Close()
+
+	var variants []string
+	var variant string
+
+	for rows.Next() {
+		err := rows.Scan(&variant)
+		if err != nil {
+			logrus.Error("Query result reading error" + err.Error())
+			return genomic_annotations.NewGetVariantsDefault(500).WithPayload(&genomic_annotations.GetVariantsDefaultBody{
+				Message: "Query result reading error " + err.Error(),
+			})
+		}
+		variants = append(variants, variant)
+	}
+
+	if len(variants) > 0 {
+		return genomic_annotations.NewGetVariantsOK().WithPayload(variants)
+	} else {
+		return genomic_annotations.NewGetVariantsNotFound()
+	}
+}
+
+func buildGetValuesQuery(params genomic_annotations.GetValuesParams) (string, error) {
+
+	if contains(genomicAnnotationsTypes, params.Annotation) {
+		return "SELECT annotation_value FROM genomic_annotations." + params.Annotation + " WHERE annotation_value ~* $1 ORDER BY annotation_value LIMIT $2", nil
```

Comment:
Yep, indeed it uses placeholders when possible. It is not possible to use placeholders for the table names though. To overcome this problem first I check if the params.Annotation value provided by the user is valid (it is contained in the list of valid genomicAnnotationTypes). If it is, I use it to build the query, otherwise I send an error message. Does it make sense to you? I did not find a better way to do this.

Reply:
Alright, I didn't know that, then indeed that's the right approach.